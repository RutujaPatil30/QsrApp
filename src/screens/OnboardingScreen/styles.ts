import { StyleSheet } from "react-native";
import theme from "../../utils/themes";
import {widthPercentageToDP as wp, heightPercentageToDP as hp} from 'react-native-responsive-screen';


const styles = StyleSheet.create({
    mainContainer:{
        flex:1,
        backgroundColor:theme.colors.primary,
        paddingVertical:hp('5%'),
        paddingHorizontal:wp('5%'),
        alignItems:'center',
    },
    logoImage:{
        height:hp("15%"),
        width:wp('30%')
    },
    ronaldImage:{
        marginTop:hp('3%'),
        height:hp("40%"),
        width:wp('80%')
    },
    title:{
        color:theme.colors.white,
        fontSize:theme.fonts.title,
        fontWeight:'bold',
    },
    subtitle:{
        marginTop:hp('3%'),
        color:theme.colors.white,
        fontSize:theme.fonts.subtitle,
        textAlign:'center'
    },
    buttonContainer:{
        flex: 1,
        justifyContent: 'flex-end',
        alignItems: 'center',
        marginBottom: hp('2%'),
        
    }
})

export default styles