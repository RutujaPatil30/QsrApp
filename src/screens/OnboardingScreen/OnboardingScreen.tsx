import React from 'react';
import { Image, SafeAreaView, Text, View } from 'react-native'
import { images } from '../../utils/images'
import styles from './styles'
import {widthPercentageToDP as wp, heightPercentageToDP as hp} from 'react-native-responsive-screen';
import baselocalization from '../../utils/baselocalization';
import CustomButton from '../../components/CustomButton/CustomButton';
import theme from '../../utils/themes';


const OnboardingScreen = ( props:any ) =>{
    const {navigation} = props;
    return(
        <SafeAreaView style={styles.mainContainer}>
            <Image source={images?.onboarding_logo}  style = {styles.logoImage} resizeMode="contain" />
            <Image  source={ images?.onboarding_ronald} style = {styles.ronaldImage}resizeMode="contain"/>
            <Text style = {styles.title}>{baselocalization.onboardingScreen.onboardingScreenTitle}</Text>
            <Text style={styles.subtitle}>{baselocalization.onboardingScreen.onboardingScreenSubtitle}</Text>
            <View style = {styles.buttonContainer}>
                <CustomButton text={baselocalization.onboardingScreen.onboardingScreenButtonText} 
                    disabled={false}
                    buttonStylesProps={{
                        backgroundColor: theme.colors.secondary,
                        borderColor: theme.colors.secondary,
                        textColor: theme.colors.primary,
                        width: wp('80%'),
                        height: hp('5%'),
                        fontSize: undefined}}
                    onPress={()=>{navigation.navigate('LoginScreen')}}
                />
            </View>
        </SafeAreaView>
    )
}

export default OnboardingScreen

