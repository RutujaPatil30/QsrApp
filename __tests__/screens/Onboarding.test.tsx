
import React from 'react';
import { render, screen } from '@testing-library/react-native';
import OnboardingScreen from '../../src/screens/OnboardingScreen/OnboardingScreen';
import { NavigationContainer, useNavigation } from '@react-navigation/native';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../src/redux/store';

describe('Checking OnboardingScreen ', () => {
  it('should render the text "OTPScreen" ', () => {

    const component = (
      <Provider store={store}>
        <NavigationContainer>
          <OnboardingScreen navigation={undefined} />
        </NavigationContainer>
      </Provider>
    );
    render(component);
    const nextButton = screen.getByTestId('OnboardingScreen_NextButton');
    expect(nextButton).toBeTruthy();
  });
});

describe('OnboardingScreen Snapshot Test', () => {
  it('renders correctly', () => {
      const tree = renderer.create(
          <Provider store={store}>
              <OnboardingScreen />
          </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
  });
});