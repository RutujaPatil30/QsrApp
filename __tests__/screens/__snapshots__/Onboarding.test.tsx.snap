// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingScreen Snapshot Test renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#da291c",
      "flex": 1,
      "paddingHorizontal": 37.5,
      "paddingVertical": 66.5,
    }
  }
>
  <RCTImageView
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    internal_analyticTag={null}
    resizeMode="contain"
    source={
      [
        {
          "height": undefined,
          "uri": undefined,
          "width": undefined,
        },
      ]
    }
    style={
      {
        "height": 200,
        "overflow": "hidden",
        "width": 225,
      }
    }
  />
  <RCTImageView
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    internal_analyticTag={null}
    resizeMode="contain"
    source={
      [
        {
          "height": undefined,
          "uri": undefined,
          "width": undefined,
        },
      ]
    }
    style={
      {
        "height": 533.5,
        "marginTop": 40,
        "overflow": "hidden",
        "width": 600,
      }
    }
  />
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 22,
        "fontWeight": "bold",
      }
    }
  >
    Selamat Datang di McDonald
  </Text>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 15,
        "marginTop": 40,
        "textAlign": "center",
      }
    }
  >
    Makin hemat dan praktis peran lewat aplikasi, banyak menu, banyak promo, nggakperlu keluar rumah!
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 26.5,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffc72c",
          "borderColor": "#ffc72c",
          "borderRadius": 53.5,
          "borderWidth": 1,
          "height": 66.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": 600,
        }
      }
      testID="OnboardingScreen_NextButton"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontSize": 18,
              "fontWeight": "bold",
            },
            {
              "color": "#da291c",
            },
          ]
        }
      >
        Pesan Sekarang
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
